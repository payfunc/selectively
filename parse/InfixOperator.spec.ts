import * as selectively from "../index"

describe("parse InfixOperator", () => {
	it("Algebraic Expressions", () => {
		expect(selectively.parseExpression("5 + 7 - 8").toString()).toEqual("5 + 7 - 8")
		expect(selectively.parseExpression("10 - 3 - 2 - 1 - 2 - 3").toString()).toEqual("10 - 3 - 2 - 1 - 2 - 3")
		expect(selectively.parseExpression("10 - 5 - 3 - 2").evaluate()).toEqual(0)
		expect(selectively.parseExpression("8 + 7 * 8 + 5").evaluate()).toEqual(69)
		expect(selectively.parseExpression("10 - 3 * 2 * 1 * 2 * 3").toString()).toEqual("10 - 3 * 2 * 1 * 2 * 3")
	})
	it("Algebraic parenthesis simple", () => {
		expect(selectively.parseExpression("(5 + 7) * 8").evaluate()).toEqual(96)
		expect(selectively.parseExpression("(5 + 7) * 8").toString()).toEqual("(5 + 7) * 8")
		expect(selectively.parseExpression("5 - 6 * (7 - 5) * 8").evaluate()).toEqual(-91)
	})
	it.skip("Algebraic parenthesis complex", () => {
		expect(selectively.parseExpression("5 - (6 * (7 - 5)) * 8").toString()).toEqual("5 - (6 * (7 - 5)) * 8")
	})
})
