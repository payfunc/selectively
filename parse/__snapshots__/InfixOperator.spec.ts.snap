// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse InfixOperator Algebraic parenthesis complex 1`] = `
InfixOperator {
  "class": "InfixOperator",
  "left": Value {
    "class": "Value",
    "name": undefined,
    "precedence": 19,
    "value": 5,
  },
  "precedence": 15,
  "right": InfixOperator {
    "class": "InfixOperator",
    "left": InfixOperator {
      "class": "InfixOperator",
      "left": Value {
        "class": "Value",
        "name": undefined,
        "precedence": 19,
        "value": 6,
      },
      "precedence": 16,
      "right": InfixOperator {
        "class": "InfixOperator",
        "left": Value {
          "class": "Value",
          "name": undefined,
          "precedence": 19,
          "value": 7,
        },
        "precedence": 15,
        "right": Value {
          "class": "Value",
          "name": undefined,
          "precedence": 19,
          "value": 5,
        },
        "symbol": "-",
      },
      "symbol": "*",
    },
    "precedence": 16,
    "right": Value {
      "class": "Value",
      "name": undefined,
      "precedence": 19,
      "value": 8,
    },
    "symbol": "*",
  },
  "symbol": "-",
}
`;
